consent-enforcement/consent-enforcement-mediator/src/main/java/org/wso2/financial/services/apim/mediation/policies/consent/enforcement/utils/ConsentEnforcementUtils.java

```diff
@@ -0,0 +1,284 @@
+        if (authHeader != null && !authHeader.isEmpty() &&
+                isValidJWTToken(authHeader.replace(ConsentEnforcementConstants.BEARER_TAG, ""))) {
+            String consentIdClaim = null;
+            if (!authHeader.contains(ConsentEnforcementConstants.BASIC_TAG)) {
+                authHeader = authHeader.replace(ConsentEnforcementConstants.BEARER_TAG, "");
+                JSONObject jwtClaims = decodeBase64(authHeader.split("\\.")[1]);
+
+                if (!jwtClaims.isNull(consentIdClaimName) && !jwtClaims.getString(consentIdClaimName).isEmpty()) {
+                    consentIdClaim = jwtClaims.getString(consentIdClaimName);
+                }
+            }
+            return consentIdClaim;
+        }
+        return null;
+    }
+
+    /**
+     * Method to create the validation request payload.
+     *
+     * @param axis2MessageContext Axis2 message context
+     * @param requestHeaders Transport headers from the Axis2 message context
+     * @param additionalParams Additional parameters to be included in the request payload
+     * @return JSONObject representing the validation request payload
+     */
+    public static JSONObject createValidationRequestPayload(org.apache.axis2.context.MessageContext axis2MessageContext,
+                                                            Map<String, Object> requestHeaders,
+                                                            Map<String, Object> additionalParams) throws JSONException {
+
+        JSONObject validationRequest = new JSONObject();
+        JSONObject headers = new JSONObject();
+
+        requestHeaders.forEach(headers::put);
+        validationRequest.put(ConsentEnforcementConstants.HEADERS_TAG, headers);
+
+        JSONObject requestPayload = new JSONObject(JsonUtil.jsonPayloadToString(axis2MessageContext));
+        validationRequest.put(ConsentEnforcementConstants.BODY_TAG, requestPayload);
+
+        additionalParams.forEach(validationRequest::put);
+        return validationRequest;
+    }
+
+    /**
+     * Method to generate JWT.
+     * @param payload Payload to be signed
+     * @return Signed JWT
+     */
+    public static String generateJWT(String payload) {
+
+        return Jwts.builder()
+                .setPayload(payload)
+                .signWith(SignatureAlgorithm.RS512, getSigningKey())
+                .compact();
+    }
+
+    /**
+     * Method to invoke consent validation service when the JWT payload is provided.
+     *
+     * @param enforcementJWTPayload JWT Payload
+     * @return Response as a String
+     * @throws IOException When failed to invoke the validation endpoint or failed to parse the response.
+     */
+    public static String invokeConsentValidationService(String enforcementJWTPayload, String consentValidationEndpoint)
+            throws IOException, APIManagementException {
+
+        HttpPost httpPost = new HttpPost(consentValidationEndpoint);
```

Comment:
Though previously thought to handle all logics in consent enforcement through custom mediator, wondering, would it ne better to use a synapse mediator to call external service instead this logic,since synapse call/send mediators already implemented..can we check that too

Reply:
Fixed with 7c52555b434bfb79d694a193d0e2124d85e8e48f